crates/ruma-common/src/api.rs

```diff
@@ -414,32 +414,34 @@ pub fn select_path<'a>(
             return Err(IntoHttpError::EndpointRemoved(metadata.removed.unwrap()))
         }
         VersioningDecision::Stable { any_deprecated, all_deprecated, any_removed } => {
-            match (any_removed, all_deprecated, any_deprecated) {
-                // Some remove, all deprecate
-                (true, true, _) => tracing::warn!(
-                        "endpoint {} is deprecated in ALL (and removed in some) of the following versions: {:?}",
+            if any_removed {
+                if all_deprecated {
+                    tracing::warn!(
+                        "endpoint {} is removed in some (and deprecated in ALL) of the following versions: {:?}",
                         metadata.name,
                         versions
-                ),
-                // Some remove, some deprecate
-                (true, false, true) => tracing::warn!(
-                    "endpoint {} is deprecated (and removed) in some versions: {:?}",
-                    metadata.name,
-                    versions
-                ),
-                // None remove, all deprecate
-                (false, true, _) => tracing::warn!(
+                    )
```

Comment:
Please put semicolons after the tracing macro calls (and maybe replace the qualified uses of `warn!` with an import).

```suggestion
                    );
```